src/modules/user/typedefs/user.ts

```diff
@@ -1,27 +1,54 @@
 import { gql } from "graphql-tag";
 
 const typedef = gql`
+  directive @auth on OBJECT
+
+  """
+  User accout info.
+  """
   type User {
-    id: String!
-    password: String!
-    last_login: String
-    is_superuser: Boolean!
+    email: String!
     username: String!
     first_name: String!
     last_name: String!
-    email: String!
+    major: [String!]!
+    last_login: String!
+    date_joined: String!
     is_staff: Boolean!
     is_active: Boolean!
-    date_joined: String!
-    major: [String],
-    email_class_update: Boolean,
-    email_grade_update: Boolean,
-    email_enrollment_opening: Boolean,
-    email_berkeleytime_update: Boolean
+    email_class_update: Boolean!
+    email_grade_update: Boolean!
+    email_enrollment_opening: Boolean!
+    email_berkeleytime_update: Boolean!
   }
 
-  type Query {
-    User(email: String!): User
+  type Query @auth {
```

Comment:
adding @auth here actually makes all queries require auth. put them on each individual query instead. same for mutations below

Reply:
Wouldn't basically all mutations require `@auth` for Users? If so, wouldn't it be repetitive to put it on each individual mutation instead of just all mutations?

Reply:
I don't anticipate any non-authenticated mutations, so if you think so as well, we could keep that on the Mutation type. But if you are going to do that, u should put it in the `common` module instead. In fact you should put the directive definition in `common` too.